anitya/lib/utilities.py

```diff
@@ -64,41 +64,44 @@ def check_project_release(project, session, test=False):
             'No backend was found for "%s"' % project.backend)
 
     publish = False
-    versions = []
-    max_version = None
-    old_version = None
+    versions_prefix = []
+    max_version = ''
 
     try:
-        versions = backend.get_versions(project)
+        versions_prefix = backend.get_versions(project)
     except exceptions.AnityaPluginException as err:
         _log.exception("AnityaError catched:")
         project.logs = str(err)
         session.add(project)
         session.commit()
         raise
 
+    # Remove prefix
+    version_class = project.get_version_class()
```

Comment:
This code is getting copied and pasted around, but it should just be a property or method on the Project model. ``project.versions`` should really be this with a ``return [str(v) for v in versions]`` so it returns a list of strings with parsing applied.

Reply:
Yeah, but this is applied before the versions are added to the project.